src/main/java/org/prebid/server/floors/BasicPriceFloorAdjuster.java

```diff
@@ -34,17 +44,69 @@ private static ExtRequestBidadjustmentfactors extractBidadjustmentfactors(BidReq
         return ObjectUtil.getIfNotNull(extPrebid, ExtRequestPrebid::getBidadjustmentfactors);
     }
 
+    private static ImpMediaType chooseMediaType(Imp imp, ExtRequestBidadjustmentfactors extBidadjustmentfactors, String bidder) {
+        if (MapUtils.isEmpty(extBidadjustmentfactors.getMediatypes())) {
+            return null;
+        }
+
+        final List<ImpMediaType> availableMediatypesForImp = retrieveImpMediatypes(imp);
+        if (CollectionUtils.isEmpty(availableMediatypesForImp)) {
+            return null;
+        }
+
+        final TreeMap<BigDecimal, ImpMediaType> factorToMediatype = new TreeMap<>();
```

Comment:
Why we need this complex logic with Map? Can we just find and return biggest floor? Then compare it with extBidadjustmentfactors.getAdjustments().get(bidder) factor?